idx/cassandra/cassandra.go

```diff
@@ -435,6 +458,42 @@ func (c *CasIdx) processWriteQueue() {
 	c.wg.Done()
 }
 
+func (c *CasIdx) addDefToArchive(def schema.MetricDefinition) error {
+	insertQry := `INSERT INTO metric_idx_archive (id, orgid, partition, name, interval, unit, mtype, tags, lastupdate, archived_at) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`
+	maxAttempts := 5
+	now := time.Now().UTC().Unix()
+	var err error
+
+	for attempts := 0; attempts < maxAttempts; attempts++ {
+		if attempts > 0 {
+			sleepTime := 100 * attempts
+			if sleepTime > 2000 {
+				sleepTime = 2000
+			}
+			time.Sleep(time.Duration(sleepTime) * time.Millisecond)
+		}
+
+		err := c.session.Query(
+			insertQry,
+			def.Id.String(),
+			def.OrgId,
+			def.Partition,
+			def.Name,
+			def.Interval,
+			def.Unit,
+			def.Mtype,
+			def.Tags,
+			def.LastUpdate,
+			now).Exec()
+
+		if err == nil {
+			return nil
+		}
```

Comment:
we need a log message if a query failed, at least every X attempts.
```suggestion
		}
		// log first failure and every 20th after that.
		if (attempts % 20) == 0 {
			log.Warnf("cassandra-idx: Failed to write def to cassandra. it will be retried. %s. the value was: %+v", err, def)
		}
```